### ISSUE TITLE
[CoE Starter Kit - QUESTION] QUESTION audit flow run fails

### ISSUE THREAD

**Issue by AlainCote** [microsoft/coe-starter-kit#9336]
### Does this question already exist in our backlog?

- [x] I have checked and confirm this is a new question.

### What is your question?

I have a similar error to this closed case but the error i am getting in the Retry logic list audit log content is different 
[CoE Starter Kit - QUESTION] Sync Audit Logs V2 fails to run with error : Get_Azure_Secret  #7250
Closed

The error code i am getting looks like a expression not being resolve or something.  I am using plain text variable not they azure key vault, so the fisrt error i understand it is not related to the faillure of the flow.

![Image](https://github.com/user-attachments/assets/074e4b66-c515-4ee8-9497-b0c3f2119331)

### What solution are you experiencing the issue with?

Core

### What solution version are you using?

november 2024 

### What app or flow are you having the issue with?

Admin | Audit Logs | Sync Audit Logs (V2)

### What method are you using to get inventory and telemetry?

Cloud flows

**Comment by Jenefer-Monroe** [comment]
Which api are you using for the audit log? 
- [Graph API](https://learn.microsoft.com/power-platform/guidance/coe/setup-auditlog-http-graphapi)
- [Office 365 Management API](https://learn.microsoft.com/power-platform/guidance/coe/setup-auditlog-http)

Have you turned on the audit logs for the tenant? [Before you set up the audit log flows](https://learn.microsoft.com/power-platform/guidance/coe/setup-auditlog-http-graphapi#before-you-set-up-the-audit-log-flows)

**Comment by AlainCote** [comment]
I am using the office 365 management API. I don't have the approriate right to validate if the audit log are turned on in the tenant but the global admin will verify this. But they are not suppose to be on by default ?

**Comment by Jenefer-Monroe** [comment]
No they are not on by default. 
Here is the easiest way to turn them on: [turn on auditing](https://learn.microsoft.com/purview/audit-log-enable-disable?preserve-view=true&view=o365-worldwide&tabs=microsoft-purview-portal#turn-on-auditing)

**Comment by AlainCote** [comment]
Hi Jenefer,

I still have the same error code even with the audit log on. ![Image](https://github.com/user-attachments/assets/71aa226b-655a-4b24-91c5-9619cb014122)

Could it be related to a environment variable not configure ? According to the documentation, the only two variables i have to configure for this flow to run, is the one for the app Id and the secret. In the termintae action i have a 500 error message,  am i missing a url somewhere ?

**Comment by Jenefer-Monroe** [comment]
Let's check your env vars. For plain text password and O365 api they should look like this:

![Image](https://github.com/user-attachments/assets/d8b681c1-c032-4351-a21f-bf4badd16412)

And then lets check your app registration. Make sure you have application, not delegated, permissions
![image](https://github.com/microsoft/coe-starter-kit/assets/89584709/bc0331a0-02d8-4306-8d70-dbe96f6f49a8)

**Comment by AlainCote** [comment]
Sorry was busy for a ehike on something else. So the audit flow is running now, it was a temporary glitch i guess. I just restarter it and it work fine now. I have a last problem with the page app usage of the report, i dont see any lauch on the visual, should i open a new ticket for this or we can continue on this one. 

![Image](https://github.com/user-attachments/assets/9d9d9f71-ce48-41ae-87f7-9efa45a28a2f)

**Comment by AlainCote** [comment]
Could my problem be link to the purview issue  #9268 ?

**Comment by Jenefer-Monroe** [comment]
Is this flow turned on? **_Admin | Audit Logs | Update Data (V2)_**

Is data populating here? 
![Image](https://github.com/user-attachments/assets/954511b4-a857-423a-9789-a95e0e25129c)

**Comment by AlainCote** [comment]
Ok so the flow is working ok. But the view you are sharing seems to be block for me, do you need global admin to see this particular view ? Do you have any documentation regarding this particular view ?

**Comment by Jenefer-Monroe** [comment]
Which view do you mean? This one? 
![Image](https://github.com/user-attachments/assets/265ec7d4-7acf-4914-b481-ebfd77e7b913)

That is the Power Platform Admin View app which is part of the CoE install.

**Comment by AlainCote** [comment]
Wow ok never thought there was a app to look at the audit inside the core component module. i can see some activities but only from 1 user on one environment (model driven on teams). Can i look in Azute / Purview / compliance portal to see if thre i more activirties that is not recorded inside that app ? 

<img width="910" alt="Image" src="https://github.com/user-attachments/assets/6cddddae-07b4-476a-b113-eabcf0b09ad4" />

**Comment by AlainCote** [comment]
No filter on the report but i can see the th flow Admin | Audit Logs | Update Data (V2) ran only once 4 days ago. What is it suppos to collect there is no description ?  Anything i should look in history of the audit flow ? 

<img width="821" alt="Image" src="https://github.com/user-attachments/assets/57756705-8226-46db-bb0b-e91018404c31" />

**Comment by Jenefer-Monroe** [comment]
This flow: Admin | Audit Logs | Update Data (V2)
runs once per audit log launch found, to update the app's last launch record

This flow: Admin | Audit Logs | Sync Audit Logs (V2)
should run hourly and collect the logs from that hour

Can you please validate that the second is running and passing hourly? 
Please launch some canvas apps, and then wait to see if those launches get recorded.

**Comment by Jenefer-Monroe** [comment]
Closing out as no further action for starter kit team
